plugins/legacy-plugin-chart-map-box/src/MapBox.jsx

```diff
@@ -108,7 +108,14 @@ class MapBox extends React.Component {
     // Compute the clusters based on the original bounds and current zoom level. Note when zoom/pan
     // to an area outside of the original bounds, no additional queries are made to the backend to
     // retrieve additional data.
-    const bbox = [bounds[0][0], bounds[0][1], bounds[1][0], bounds[1][1]];
+    // add this variable to widen the visible area
+    const offset = 0.05;
+    const bbox = [
+      bounds[0][0] - offset,
+      bounds[0][1] - offset,
+      bounds[1][0] + offset,
+      bounds[1][1] + offset,
+    ];
```

Comment:
Applying a constant offset of 0.05 could be much or little depending on where on the zoom level. Would it make sense to make this a percentage of the width for the horizontal and height for the vertical bounds?

Reply:
done, 0.5 percentage of width and height. Can you check, please?